src/cascade/model/rates.py

```diff
@@ -0,0 +1,76 @@
+class Smooth:
+    __slots__ = ["_value_priors", "_d_age_priors", "_d_time_priors"]
+
+    def __init__(self, value_priors=None, d_age_priors=None, d_time_priors=None):
+        self._value_priors = None
+        self._d_age_priors = None
+        self._d_time_priors = None
+
+        self.value_priors = value_priors
+        self.d_age_priors = d_age_priors
+        self.d_time_priors = d_time_priors
+
+    def _validate_grids(self, priors):
+        grids = [ps.grid for ps in [self.value_priors, self.d_age_priors, self.d_time_priors, priors] if ps]
+        if grids:
+            if not all([grids[0] == g for g in grids]):
+                raise ValueError("Smooth cannot contain priors on hetrogenious grids")
```

Comment:
ok, I like the idea of "genius" grids.

maybe this should be: heterogeneous ?

Reply:
Yesyes